Kernel/Storage/AHCIController.cpp

```diff
@@ -79,13 +79,14 @@ volatile AHCI::HBA& AHCIController::hba() const
     return static_cast<volatile AHCI::HBA&>(*(volatile AHCI::HBA*)(m_hba_region->vaddr().as_ptr()));
 }
 
-AHCIController::AHCIController(PCI::Address address)
+AHCIController::AHCIController(PCI::Address address, bool init)
     : StorageController()
     , PCI::DeviceController(address)
     , m_hba_region(hba_region())
     , m_capabilities(capabilities())
 {
-    initialize();
+    if (init)
```

Comment:
Maybe instead of passing a boolean parameter, we could make `initialize` a virtual function so we can override it. In that  case, we will not call initialize method from the constructor because it will use the method from the parent class and not from the derived class.